script/testing/micro_bench/artifact_processor.py

```diff
@@ -0,0 +1,165 @@
+                  'status', 'iterations', 'ref_throughput', 'num_results']
+
+        publishable_comparison = {
+            key: value for key, value in initial_comparison.items() if key in fields}
+
+        test_suite, test_name = initial_comparison.get('suite'), initial_comparison.get('test')
+
+        key = (gbench_result.suite_name, gbench_result.test_name)
+        historical_results = self.artifacts.get(key)
+        if initial_comparison.get('reference_type') != 'none':
+            publishable_comparison['stdev_throughput'] = historical_results.get_stdev_throughput()
+
+        return publishable_comparison
+
+    def get_comparison(self, bench_name, gbench_result, lax_tolerance=None):
+        """ create and return a dict comparing the historical benchmark results
+            to the benchmark result passed in
+        """
+        key = (gbench_result.suite_name, gbench_result.test_name)
+        historical_results = self.artifacts.get(key, None)
+        ref_type = 'none'
+        if historical_results:
+            if historical_results.get_num_results() >= self.required_num_results:
+                ref_type = 'historic'
+            else:
+                ref_type = 'lax'
+        return self._create_comparison_dict(bench_name, gbench_result, ref_type, lax_tolerance)
+
+    def _create_comparison_dict(self, bench_name, gbench_result, ref_type='none', lax_tolerance=None):
+        """ creates the comparison dict based on the ref type. It takes a
+        GBenchTestResult and compares it against the results for the same 
+        benchmark stored in the artifact processor. """
+        key = (gbench_result.suite_name, gbench_result.test_name)
+        LOG.debug("Loading {REF_TYPE} history data for {SUITE_NAME}.{TEST_NAME} [{BENCH_NAME}]".format(
+            REF_TYPE=ref_type, SUITE_NAME=gbench_result.suite_name, TEST_NAME=gbench_result.test_name, BENCH_NAME=bench_name
+        ))
+        comparison = {
+            "suite": gbench_result.suite_name,
+            "test": gbench_result.test_name,
+            "num_results": 0,
+            "throughput": gbench_result.items_per_second,
+            "iterations": gbench_result.iterations,
+            "tolerance": 0,
+            "reference_type": ref_type,
+            "status": "PASS"
+        }
+        if ref_type != 'none':
+            historical_results = self.artifacts.get(key)
+            comparison['num_results'] = historical_results.get_num_results()
+            comparison['tolerance'] = BENCHMARKS_TO_RUN[bench_name] if ref_type == 'historic' else lax_tolerance
+            comparison['ref_throughput'] = historical_results.get_mean_throughput()
+            comparison['coef_var'] = 100 * historical_results.get_stdev_throughput() / comparison.get('ref_throughput')
+            comparison['change'] = 100 * (gbench_result.items_per_second -
+                                          comparison.get('ref_throughput')) / comparison.get('throughput')
```

Comment:
Can we ensure that `gbench_result.items_per_second` always returns a non-`None` and positive int value? Because otherwise the `comparison['change']` calculation can be prone to `ValueError`.